Automattic/jetpack
# REST requests: avoid PHP warnings with custom post objects

Fixes #33728
Fixes #36306

## Proposed changes:

In some scenarios, the object data passed to the callback function used with `register_rest_field` is not the standard `WP_Post` post data we can expect. In order to avoid PHP warnings like the ones listed in the issues above, let's be more defensive everywhere where we use `register_rest_field`, and bail early when we do not have the expected data. 
### Other information:


## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

I'm not exactly sure what's the best way to reproduce the warnings above, since it depends on specific third-party plugins. I think we can test things out by ensuring that the different requests still work for a regular setup.

* Go to Jetpack > Settings > Sharing
* Enable sharing buttons and Publicize
* Go to Posts > Add New
* Publish a post
    * On the frontend, ensure you can see sharing buttons.
* Go back to the post editor, and in the Jetpack sidebar disable sharing.
* Save your changes
    * On the frontend, ensure the buttons are now gone.
* Go to Posts > Add New
* Connect a few social networks.
* Write a post
* Unselect some of the social networks you've connected in the Jetpack sidebar
* Publish your post
    * Ensure the posts are only sent to the selected social networks.
* Go to Jetpack > Settings > Performance
* Turn on VideoPress
* Go to Posts > Add New
* Add a VideoPress block and upload a video.
* Make sure the video can be uploaded, and its details changed.
